whois-update/src/main/java/net/ripe/db/whois/update/domain/UpdateMessages.java

```diff
@@ -656,6 +656,18 @@ public static Message multipleUserMntBy(final Collection<CIString> userMntners)
         return new Message(Type.ERROR, "Multiple user-'mnt-by:' are not allowed, found are: '%s'", Joiner.on(", ").join(userMntners));
     }
 
+    public static Message notValidSource() {
+        return new Message(Type.ERROR, "Can not set NONAUTH source when aut-num is not NONAUTH");
+    }
+
+    public static Message flatModelNotAllowSourceModifications(final String source){
+        return new Message(Type.ERROR, "Can not set '%s' source when as-set in not hierarchical", source);
```

Comment:
Do we need a check for flat as-set?

Reply:
I am changing the error message, now it will be a warning and it will explain that the update request will change because we are not allow to update the source, we will take the database's source (this warning can only be thrown for update)

Reply:
But the check or warning is only for hierarchal for the flat there should not be any change in source?

Reply:
in the jira task: 
Do not update the source of flat AS-SET names (i.e. without an AUT-NUM reference).

We can not update the source, so if the user updates the source we need to notify about the change I guess

Reply:
There's no AUT-NUM reference in flat AS-SET names (by definition) so no need to add any warning, just do nothing.